### PR TITLE
CLDR-15034 kbd: move pcm keyboard to 3.0 format

### PR DESCRIPTION
**Pull request by srl295**
CLDR-15034

- [ ] This PR completes the ticket.

Conversion of the `pcm` keyboard from pre-3.0 to new format.

image

The prior format file is here: https://github.com/unicode-org/cldr/blob/maint/maint-43/keyboards/und/pcm-t-k0-windows.xml

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:68]: This is a peculiar order of transforms (and with the `''` thrown in the middle), is that deliberate? It is difficult to see whether all base chars or all combining marks are covered.

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:38]: why new row? (also in the layer above)

Comment (srl295) [reply keyboards/3.0/pcm.xml:38]: Why the new line? Just standard redhat xml formatter. I'll fix it.

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:7]: Yeah even though we said the implementation can pick a name it wants (or the first one), I feel like it would be useful to somehow note down what these values are, or how they differ, similar to `rel` attribute in `link`

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:43]: Shouldn't there also be a caps layer? I don't see in the spec preview that that would happen automatically.

Comment (srl295) [reply keyboards/3.0/pcm.xml:43]: Yes, there should, and there is one in the original file

Comment (srl295) [reply keyboards/3.0/pcm.xml:43]: Good catch

Comment (miloush) [reply keyboards/3.0/pcm.xml:43]: Maybe we should say something like unless there is an explicit caps layer, caps case-maps everything to uppercase

Comment (srl295) [reply keyboards/3.0/pcm.xml:43]: It could be a tool feature too though, generate a caps layer based on unicode casing.

**Comment by srl295** [comment]
Can you file a ticket on that ?

File: keyboards/3.0/pcm.xml
Comment (mcdurdin) [reply keyboards/3.0/pcm.xml:43]: Leave it to tooling I think -- auto caps layer adds complexity to implementations and necessitates a bunch of orthography assumptions which would be problematic and could lead to frustrating implementation inconsistencies and bugs. Turkish i/ı/İ/I is just the beginning.

File: keyboards/3.0/pcm.xml
Comment (mcdurdin) [reply keyboards/3.0/pcm.xml:7]: :+1: `rel` attribute would be very helpful.

File: keyboards/3.0/pcm.xml
Comment (mcdurdin) [reply keyboards/3.0/pcm.xml:68]: These transforms (apart from `''`) are just normalization, and should not be included as part of the keyboard -- it should be automatic per the spec.

File: keyboards/3.0/pcm.xml
Comment (miloush) [reply keyboards/3.0/pcm.xml:43]: Well the layout does specify a locale that could be used for the case mapping, although we also support noting down alternative locales. The conflicting goal here is to have minimal files. I am perfectly happy with leaving this to tooling rather than the implementation.

File: keyboards/3.0/pcm.xml
Comment (miloush) [reply keyboards/3.0/pcm.xml:68]: Didn't we say the specified normalization is descriptive and does not change the output in any way?

Comment (srl295) [reply keyboards/3.0/pcm.xml:68]: Correct, my understanding is that the normalization is descriptive.

File: keyboards/3.0/pcm.xml
Comment (srl295) [reply keyboards/3.0/pcm.xml:7]: great idea, file a ticket ? :)

File: keyboards/3.0/pcm.xml
Comment (srl295) [reply keyboards/3.0/pcm.xml:43]: > locale used for case mapping

That's very much locale data, unicode version specific.  yes, that does make the files a little less minimal, but once the file is there it's completely independent of all Unicode versions and all locale data situations, which is best for less user surprise. As probably a bad example but Cherokee and Georgian didn't originally have bicamerality and now they do, so caps behavior could change dramatically.

File: keyboards/3.0/pcm.xml
Comment (mcdurdin) [reply keyboards/3.0/pcm.xml:68]: Ticket relating to this: https://unicode-org.atlassian.net/browse/CLDR-16943

File: keyboards/3.0/pcm.xml
Comment (srl295) [reply keyboards/3.0/pcm.xml]: @miloush do you want to file a ticket on this one?

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:55]: Is this apos? I wonder how intuitive it is to have all these as implicit import

**Comment by miloush** [comment]
Is there are ground truth for this particular layout, like a standard or community website or something that could be referenced/checked against?

File: keyboards/3.0/pcm.xml
Comment (srl295) [reply keyboards/3.0/pcm.xml:55]: This isn't an implicit import. this is a double apostrophe, as in the original file.

**Comment by srl295** [comment]
this will have to be rebased once #3274  lands.

@miloush any other concerns with it?  Please compare to the original file: https://github.com/unicode-org/cldr/blob/maint/maint-43/keyboards/und/pcm-t-k0-windows.xml

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:38]: The original file has a `question` at the end of this line, like the other states have `slash`
```suggestion
      <row keys="A S D F G H J K L Odot Edot question" />
```

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:44]: As far as I can tell, the original file has combining grave in _none_ and _shift_ states and non-combinign one here. In this PR all seem to be the same.

File: keyboards/3.0/pcm.xml
Comment (miloush) [inline keyboards/3.0/pcm.xml:45]: The original file has Q for _caps_ state
```suggestion
      <row keys="Q W E R T Y U I O P open-square close-square" />
```

**Comment by miloush** [comment]
Let me know if these were intentional differences.

**Comment by srl295** [comment]
@miloush i will get back to you on these… for now, this will need revamping also because it's now out of date due to the DTD

File: keyboards/3.0/pcm.xml
Comment (srl295) [reply keyboards/3.0/pcm.xml:44]: good catch, fixed

**Comment by srl295** [comment]
@miloush  and all PTAL

**Review by miloush — APPROVED** [review]

**Pull request merged**
